src/bufr/BufrParser/BufrCollectors/BufrIntCollector.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * (C) Copyright 2020 NOAA/NWS/NCEP/EMC
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+#include "bufr.interface.h"
+
+#include "BufrIntCollector.h"
+#include "BufrParser/BufrMnemonicSet.h"
+
+
+namespace Ingester
+{
+    BufrIntCollector::BufrIntCollector(const int fileUnit, const BufrMnemonicSet mnemonicSet) :
+        BufrCollector(fileUnit,
+                      BufrAccumulator(mnemonicSet.getSize() * mnemonicSet.getMaxColumn())),
+        mnemonicSet_(mnemonicSet)
+    {
+        scratchData_ = new double[mnemonicSet.getSize() * mnemonicSet.getMaxColumn()];
```

Comment:
Use a std::vector<double> instead of trying to manage your memory manually.

Reply:
Done